qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -142,57 +130,97 @@ def __init__(self,
             initial_point = var_form.preferred_init_points
 
         self._max_evals_grouped = max_evals_grouped
-
-        self._in_operator = None
-        self._operator = None
-        self._in_aux_operators = None
-        self._aux_operators = None
-        self._callback = callback
-        self._auto_conversion = auto_conversion
-
-        self._use_simulator_snapshot_mode = None
-        self._ret = None
-        self._eval_time = None
-        self._eval_count = 0
+        self._circuit_sampler = None
+        self._expect_op = None
 
         super().__init__(var_form=var_form,
                          optimizer=optimizer,
                          cost_fn=self._energy_evaluation,
                          initial_point=initial_point,
                          quantum_instance=quantum_instance)
+        self._ret = None
+        self._eval_time = None
+        self._optimizer.set_max_evals_grouped(max_evals_grouped)
+        self._callback = callback
 
-        logger.info(self.print_settings())
-        self._parameterized_circuits = None
+        self._expectation = expectation
+        self._operator = None
+        if operator is not None:
+            self.operator = operator
+        self.aux_operators = aux_operators
 
-        self.operator = operator
-        aux_ops = []
-        if aux_operators is not None:
-            aux_operators = \
-                [aux_operators] if not isinstance(aux_operators, list) else aux_operators
-            for aux_op in aux_operators:
-                aux_ops.append(aux_op)
-        self.aux_operators = aux_ops
+        self._eval_count = 0
+        logger.info(self.print_settings())
 
     @property
-    def operator(self) -> Optional[BaseOperator]:
+    def operator(self) -> Optional[OperatorBase]:
         """ Returns operator """
-        return self._in_operator
+        return self._operator
 
     @operator.setter
-    def operator(self, operator: BaseOperator) -> None:
+    def operator(self, operator: Union[OperatorBase, LegacyBaseOperator]) -> None:
         """ set operator """
-        self._in_operator = operator
+        if isinstance(operator, LegacyBaseOperator):
+            operator = operator.to_opflow()
+        self._operator = operator
+        self._expect_op = None
         self._check_operator_varform()
+        if self._expectation is None:
+            self._try_set_expectation_value_from_factory()
+
+    def _try_set_expectation_value_from_factory(self):
+        if self.operator and self.quantum_instance:
+            self.expectation = ExpectationFactory.build(operator=self.operator,
+                                                        backend=self.quantum_instance)
+
+    @QuantumAlgorithm.quantum_instance.setter
+    def quantum_instance(self, quantum_instance: Union[QuantumInstance, BaseBackend]) -> None:
```

Comment:
This would be preferable if it call the setter it os overridding to do its part first then did after what was needed specifically here in this class, rather than replicating code from Quantum Algo here

Reply:
Fixed!